auto_swap_initiator.js

```diff
@@ -1,111 +1,112 @@
-    var transactionsData = await transactionsResponse.json();
-
-    var transactions = transactionsData.data;
-    if( transactions.length == 0) {
-        console.log("No more transactions to process");
-        pullMore = false;
-        break;
+    var walletResponse = await fetch("https://api.shakepay.com/wallets", { "headers": { "accept": "application/json", "accept-language": "en-US,en;q=0.9,fr;q=0.8", "authorization": window.sessionStorage.getItem("feathers-jwt"), "cache-control": "no-cache", "content-type": "application/json", "pragma": "no-cache", "sec-ch-ua": "\" Not A;Brand\";v=\"99\", \"Chromium\";v=\"90\", \"Microsoft Edge\";v=\"90\"", "sec-ch-ua-mobile": "?0", "sec-fetch-dest": "empty", "sec-fetch-mode": "cors", "sec-fetch-site": "same-site" }, "referrerPolicy": "same-origin", "body": null, "method": "GET", "mode": "cors", "credentials": "include" });
+    var wallets = await walletResponse.json();
+    for (let i in wallets.data) {
+        if (wallets.data[i].currency == "CAD") {
+            wallet = wallets.data[i];
+            break;
+        }
     }
 
-    for (var i = 0; i < transactions.length; i++) {
-        var t = transactions[i];                
+    var localTime = new Date();
+    var localTimeOffset = localTime.getTimezoneOffset();
+    var msOffset = (localTimeOffset - 240) * 60 * 1000;
+    var easternTime = new Date(localTime.getTime() + msOffset);
+    var midnightStart = new Date(easternTime.getFullYear(), easternTime.getMonth(), easternTime.getDate(), 0, 0, 0, 0);
+    var startTime = new Date(midnightStart.getTime() - msOffset);
 
-        if(typeof transactionCatalog[t.transactionId] != 'undefined') continue;
-        transactionCatalog[t.transactionId] = 1;            
+    var pullMore = true;
+    var swapperBalance = [];
+    var transactionCatalog = [];
+    var page = 1;
 
-        if(t.type!="peer") continue;
-        if(t.currency!="CAD") continue;
+    while (pullMore === true) {
+        console.log("pulling 2000 transactions from Shakepay api");
+        var transactionsResponse = await fetch("https://api.shakepay.com/transactions/history", { "headers": { "accept": "application/json", "accept-language": "en-US,en;q=0.9,fr;q=0.8", "authorization": window.sessionStorage.getItem("feathers-jwt"), "content-type": "application/json", "sec-ch-ua": "\" Not A;Brand\";v=\"99\", \"Chromium\";v=\"90\", \"Microsoft Edge\";v=\"90\"", "sec-ch-ua-mobile": "?0", "sec-fetch-dest": "empty", "sec-fetch-mode": "cors", "sec-fetch-site": "same-site" }, "referrerPolicy": "same-origin", "body": "{\"pagination\":{\"descending\":true,\"rowsPerPage\":2000,\"page\":" + page + "},\"filterParams\":{}}", "method": "POST", "mode": "cors", "credentials": "include" })
+        var transactionsData = await transactionsResponse.json();
 
-        createdAt = parseInt(Date.parse(t.createdAt));
-        if(createdAt < startTime.getTime()) {
+        var transactions = transactionsData.data;
+        if (transactions.length == 0) {
```

Comment:
why did you do that? this is supposed to stop loading when it reaches transactions from the day before.

Reply:
nevermind.